## Data Analysis Project

1. Which state has the high number of orders
2. According to data in which year sale was high.
3. Payment Methods that are used widely.
4. Top Products that make highest sales with payment method.
5. Which state has contributed most in the sale.

##### Top States with the high number of orders

Importing the important libraries

In [68]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
from scipy import stats
%matplotlib inline

Reading the CSV file

In [3]:
data = pd.read_csv("D:\Rahul\Internship\dataset\orders_2020_2021_DataSet_Updated.csv")

Showing the Top 5 rows of dataset

In [4]:
data.head()

Order # Fulfillment Date and Time Stamp Currency Subtotal  \
0  R121113121                             NaN      INR     2299   
1  R472890631                             NaN      INR     2299   
2  R004476488                             NaN      INR     2299   
3  R526038353                             NaN      INR     2299   
4  R658530771                             NaN      INR      349   

  Shipping Method  Shipping Cost Tax Method  Taxes Total Coupon Code  ...  \
0      Ships Free              0        NaN    0.0  2299         NaN  ...   
1      Ships Free              0        NaN    0.0  2299         NaN  ...   
2      Ships Free              0        NaN    0.0  2299         NaN  ...   
3      Ships Free              0        NaN    0.0  2299         NaN  ...   
4      Ships Free              0        NaN    0.0   349         NaN  ...   

  Tracking #  Special Instructions  \
0        NaN                   NaN   
1        NaN                   NaN   
2        NaN                   NaN   
3        NaN                   NaN   
4        NaN                   NaN   

                                       LineItem Name LineItem SKU  \
0  Falcon Drone Four Axis Aircraft with 2.4 GHz R...    PL0093416   
1  Falcon Drone Four Axis Aircraft with 2.4 GHz R...    PL0093416   
2  Falcon Drone Four Axis Aircraft with 2.4 GHz R...    PL0093416   
3  Falcon Drone Four Axis Aircraft with 2.4 GHz R...    PL0093416   
4                                 Barbie Doll (pink)      PL00098   

  LineItem Options LineItem Add-ons LineItem Qty LineItem Sale Price  \
0              NaN              NaN            1                2299   
1              NaN              NaN            1                2299   
2              NaN              NaN            1                2299   
3              NaN              NaN            1                2299   
4              NaN              NaN            1                 349   

  Download Status LineItem Type  
0             NaN      physical  
1             NaN      physical  
2             NaN      physical  
3             NaN      physical  
4             NaN      physical  

[5 rows x 38 columns]

In [5]:
data.columns

Index(['Order #', 'Fulfillment Date and Time Stamp', 'Currency', 'Subtotal',
       'Shipping Method', 'Shipping Cost', 'Tax Method', 'Taxes', 'Total',
       'Coupon Code', 'Coupon Code Name', ' ', 'Billing Name',
       'Billing Country', 'Billing Street Address', 'Billing Street Address 2',
       'Billing City', 'Billing State', 'Billing Zip', 'Shipping Name',
       'Shipping Country', 'Shipping Street Address',
       'Shipping Street Address 2', 'Shipping City', 'Shipping State',
       'Shipping Zip', 'Gift Cards', 'Payment Method', 'Tracking #',
       'Special Instructions', 'LineItem Name', 'LineItem SKU',
       'LineItem Options', 'LineItem Add-ons', 'LineItem Qty',
       'LineItem Sale Price', 'Download Status', 'LineItem Type'],
      dtype='object')

Gathering the Columns that are needed

In [6]:
data = data[['Order #', 'Fulfillment Date and Time Stamp', 'Currency', 'Total', 'Shipping Country',
        'Shipping City', 'Shipping State', 'Payment Method', 'LineItem Name', 'LineItem Qty']]

In [7]:
data.head()

Order # Fulfillment Date and Time Stamp Currency Total Shipping Country  \
0  R121113121                             NaN      INR  2299              IND   
1  R472890631                             NaN      INR  2299              IND   
2  R004476488                             NaN      INR  2299              IND   
3  R526038353                             NaN      INR  2299              IND   
4  R658530771                             NaN      INR   349              IND   

  Shipping City Shipping State Payment Method  \
0    Chandrapur          IN-MH            NaN   
1    Chandrapur          IN-MH            NaN   
2    Chandrapur          IN-MH            NaN   
3    Chandrapur          IN-MH            NaN   
4     NEW DELHI          IN-DL            NaN   

                                       LineItem Name  LineItem Qty  
0  Falcon Drone Four Axis Aircraft with 2.4 GHz R...             1  
1  Falcon Drone Four Axis Aircraft with 2.4 GHz R...             1  
2  Falcon Drone Four Axis Aircraft with 2.4 GHz R...             1  
3  Falcon Drone Four Axis Aircraft with 2.4 GHz R...             1  
4                                 Barbie Doll (pink)             1

Changing the column name to the standard need

In [8]:
data.rename(columns={'Order #':'Order No', 'Fulfillment Date and Time Stamp':'DateTime Stamp', 'LineItem Name':'Product Name',
        'LineItem Qty':'Quantity'}, inplace=True)

In [9]:
data.columns

Index(['Order No', 'DateTime Stamp', 'Currency', 'Total', 'Shipping Country',
       'Shipping City', 'Shipping State', 'Payment Method', 'Product Name',
       'Quantity'],
      dtype='object')

In [10]:
data.shape

(3039, 10)

Handling the Duplicate records in the dataset

In [11]:
data.duplicated().sum()

3

In [12]:
data.drop_duplicates()

Order No             DateTime Stamp Currency Total Shipping Country  \
0     R121113121                        NaN      INR  2299              IND   
1     R472890631                        NaN      INR  2299              IND   
2     R004476488                        NaN      INR  2299              IND   
3     R526038353                        NaN      INR  2299              IND   
4     R658530771                        NaN      INR   349              IND   
...          ...                        ...      ...   ...              ...   
3034  R243395005                        NaN      INR    50              IND   
3035  R607209508                        NaN      INR   500              IND   
3036  R140316623                        NaN      INR  4000              IND   
3037  R192037084                        NaN      INR  2000              IND   
3038  R414254148  28-09-2016 19:05:30 +0530      INR  2000              IND   

     Shipping City Shipping State             Payment Method  \
0       Chandrapur          IN-MH                        NaN   
1       Chandrapur          IN-MH                        NaN   
2       Chandrapur          IN-MH                        NaN   
3       Chandrapur          IN-MH                        NaN   
4        NEW DELHI          IN-DL                        NaN   
...            ...            ...                        ...   
3034         Gccch          IN-AN                        NaN   
3035     BERHAMPUR          IN-OR                        NaN   
3036     berhampur          IN-OR                        NaN   
3037          sdas          IN-AN                        NaN   
3038          sdas          IN-AN  Offline Payment ₹2,000.00   

                                           Product Name  Quantity  
0     Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
1     Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
2     Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
3     Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
4                                    Barbie Doll (pink)         1  
...                                                 ...       ...  
3034                 Ear Wired Earphones With Mic White         1  
3035                                  I Kall K-66 White         1  
3036                  Electric Steam Cooker Maestro MC1         2  
3037                  Electric Steam Cooker Maestro MC1         1  
3038                  Electric Steam Cooker Maestro MC1         1  

[3036 rows x 10 columns]

Result : After dropping 3 duplicate values now there is 3036 records in dataset.

Checking the null values in each column

In [13]:
data.isnull().sum()

Order No               0
DateTime Stamp      2801
Currency               0
Total                  0
Shipping Country       0
Shipping City         19
Shipping State        22
Payment Method      2780
Product Name           0
Quantity               0
dtype: int64

Checking the percentage(%) of missing values in columns

In [14]:
data.isnull().mean()

Order No            0.000000
DateTime Stamp      0.921685
Currency            0.000000
Total               0.000000
Shipping Country    0.000000
Shipping City       0.006252
Shipping State      0.007239
Payment Method      0.914775
Product Name        0.000000
Quantity            0.000000
dtype: float64

Checking the number of unique value in each column

In [15]:
data.nunique()

Order No            2971
DateTime Stamp       203
Currency               1
Total                249
Shipping Country      21
Shipping City       1376
Shipping State        52
Payment Method       106
Product Name         270
Quantity              24
dtype: int64

In [16]:
data.head()

Order No DateTime Stamp Currency Total Shipping Country Shipping City  \
0  R121113121            NaN      INR  2299              IND    Chandrapur   
1  R472890631            NaN      INR  2299              IND    Chandrapur   
2  R004476488            NaN      INR  2299              IND    Chandrapur   
3  R526038353            NaN      INR  2299              IND    Chandrapur   
4  R658530771            NaN      INR   349              IND     NEW DELHI   

  Shipping State Payment Method  \
0          IN-MH            NaN   
1          IN-MH            NaN   
2          IN-MH            NaN   
3          IN-MH            NaN   
4          IN-DL            NaN   

                                        Product Name  Quantity  
0  Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
1  Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
2  Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
3  Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
4                                 Barbie Doll (pink)         1

1. Which States of india have the high sales.

In [17]:
india_country = data[data['Shipping Country'] == 'IND']


In [18]:
top_state = india_country[['Shipping State', 'Shipping Country']]
top_state.head()

Shipping State Shipping Country
0          IN-MH              IND
1          IN-MH              IND
2          IN-MH              IND
3          IN-MH              IND
4          IN-DL              IND

In [19]:
top_state['Count States'] = india_country.groupby(['Shipping State'])['Order No'].transform('count')
top_state.head()

C:\Users\Admin\AppData\Local\Temp\ipykernel_11140\4150116577.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  top_state['Count States'] = india_country.groupby(['Shipping State'])['Order No'].transform('count')


Shipping State Shipping Country  Count States
0          IN-MH              IND           379
1          IN-MH              IND           379
2          IN-MH              IND           379
3          IN-MH              IND           379
4          IN-DL              IND           129

In [20]:
top_state.drop(['Shipping Country'], axis=1, inplace=True)
top_state.columns

C:\Users\Admin\AppData\Local\Temp\ipykernel_11140\4002702850.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  top_state.drop(['Shipping Country'], axis=1, inplace=True)


Index(['Shipping State', 'Count States'], dtype='object')

In [21]:
top_state['Count States'].astype('int64')

0       379
1       379
2       379
3       379
4       129
       ... 
3034    155
3035     80
3036     80
3037    155
3038    155
Name: Count States, Length: 2999, dtype: int64

In [22]:
top_10_states = top_state.groupby(['Shipping State'], as_index=False)['Count States'].median().sort_values(by='Count States', ascending=False)
top_10_states = top_10_states.head(10)
top_10_states.reset_index(inplace=True, drop=True)
top_10_states.head()

Shipping State  Count States
0          IN-MH         379.0
1          IN-TN         356.0
2          IN-UP         289.0
3          IN-KA         205.0
4          IN-AP         178.0

In [23]:
top_10_states

Shipping State  Count States
0          IN-MH         379.0
1          IN-TN         356.0
2          IN-UP         289.0
3          IN-KA         205.0
4          IN-AP         178.0
5          IN-TG         159.0
6          IN-AN         155.0
7          IN-DL         129.0
8          IN-BR         123.0
9          IN-WB         120.0

Replacing the values of [Shipping State] to the understandable format

In [24]:
top_10_states['Shipping State'] = top_10_states['Shipping State'].replace(['IN-MH', 'IN-TN', 'IN-UP', 'IN-KA', 'IN-AP', 'IN-TG', 'IN-AN', 'IN-DL', 'IN-BR', 'IN-WB'],
        ['Maharashtra', 'Tamil Nadu', 'Uttar Pradesh', 'Karnataka', 'Andhra Pradesh', 'Telangana', 'Andaman and Nicobar Islands',
        'Delhi', 'Bihar', 'West Bengal'])
top_10_states.head()

Shipping State  Count States
0     Maharashtra         379.0
1      Tamil Nadu         356.0
2   Uttar Pradesh         289.0
3       Karnataka         205.0
4  Andhra Pradesh         178.0

Visualizing the Results of 'Sales By States'

In [25]:
state_plot = px.bar(top_10_states, x='Shipping State', y='Count States', title='Top 10 Number of Sales By States',
        labels={'Shipping State':'State', 'Count States':'Count of Orders'}, text_auto=True)
state_plot.show()

##### Results - As per the Visuals we can see that Maharashtra State has the high number of orders.

##### 2) Checking the sale according to Yearly.

Filtering the non-nan values in the dataset.

In [26]:
year_sale_data = data[data['DateTime Stamp'].notnull()]
year_sale_data.head()

Order No             DateTime Stamp Currency Total Shipping Country  \
15   R679506806  23-09-2021 18:09:27 +0530      INR  1000              IND   
18   R444302271  23-09-2021 18:39:07 +0530      INR   499              IND   
21   R181106661  23-09-2021 18:39:50 +0530      INR   499              IND   
254  R798254987  27-04-2021 15:08:07 +0530      INR   321              IND   
292  R299286513  05-04-2021 22:50:28 +0530      INR   999              IND   

    Shipping City Shipping State Payment Method  \
15       VARANASI          IN-UP            NaN   
18          Delhi          IN-DL            NaN   
21          Delhi          IN-DL            NaN   
254     Berhampur          IN-OR            NaN   
292     Hyderabad          IN-AP            NaN   

                                          Product Name  Quantity  
15   ONLINE TUITION FOR STD 4 and STD 5(FREE 5 DAYS...         1  
18   Fun Activities for Kids LKG to STD 10(Singing,...         1  
21   Fun Activities for Kids LKG to STD 10(Singing,...         1  
254                 Mutton Curry Cut 500gm (Berhampur)         1  
292  REALME BUDS PLUS Wireless Bluetooth Ears Buds ...         1

In [27]:
year_sale_data.shape

(238, 10)

In [28]:
year_sale_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 238 entries, 15 to 3038
Data columns (total 10 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Order No          238 non-null    object
 1   DateTime Stamp    238 non-null    object
 2   Currency          238 non-null    object
 3   Total             238 non-null    object
 4   Shipping Country  238 non-null    object
 5   Shipping City     237 non-null    object
 6   Shipping State    238 non-null    object
 7   Payment Method    233 non-null    object
 8   Product Name      238 non-null    object
 9   Quantity          238 non-null    int64 
dtypes: int64(1), object(9)
memory usage: 20.5+ KB


In [29]:
year_sale_data['DateTime Stamp'] = year_sale_data['DateTime Stamp'].astype('datetime64[ns]')
year_sale_data.dtypes

C:\Users\Admin\AppData\Local\Temp\ipykernel_11140\2800283185.py:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



Order No                    object
DateTime Stamp      datetime64[ns]
Currency                    object
Total                       object
Shipping Country            object
Shipping City               object
Shipping State              object
Payment Method              object
Product Name                object
Quantity                     int64
dtype: object

In [30]:
year_sale_data['Total']=year_sale_data['Total'].astype('float64')
year_sale_data['Total'] = year_sale_data['Total'].astype('int64')
year_sale_data.dtypes

C:\Users\Admin\AppData\Local\Temp\ipykernel_11140\1181335453.py:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

C:\Users\Admin\AppData\Local\Temp\ipykernel_11140\1181335453.py:2: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



Order No                    object
DateTime Stamp      datetime64[ns]
Currency                    object
Total                        int64
Shipping Country            object
Shipping City               object
Shipping State              object
Payment Method              object
Product Name                object
Quantity                     int64
dtype: object

In [31]:
year_sale_data['Year'] = pd.to_datetime(year_sale_data['DateTime Stamp']).dt.year

C:\Users\Admin\AppData\Local\Temp\ipykernel_11140\759780284.py:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



Handling the dupicate values for better results.

In [32]:
year_sale_data.duplicated().sum()

2

In [33]:
year_sale_data[year_sale_data.duplicated()]

Order No      DateTime Stamp Currency  Total Shipping Country  \
2558  R754403637 2019-03-28 19:29:45      INR  44970              IND   
2614  R446418996 2019-05-31 20:03:48      INR  46166              IND   

     Shipping City Shipping State              Payment Method  \
2558      Varanasi          IN-UP  Offline Payment ₹44,970.00   
2614      Varanasi          IN-UP  Offline Payment ₹46,166.00   

                                           Product Name  Quantity  Year  
2558  Vmax HX 750 Remote Control Flying Drone No Camera        15  2019  
2614  Vmax HX 750 Remote Control Flying Drone No Camera         5  2019

In [34]:
year_sale_data.drop_duplicates(inplace=True)
year_sale_data.duplicated().sum()

C:\Users\Admin\AppData\Local\Temp\ipykernel_11140\2456456962.py:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



0

In [35]:
year_sale_data.head()

Order No      DateTime Stamp Currency  Total Shipping Country  \
15   R679506806 2021-09-23 12:39:27      INR   1000              IND   
18   R444302271 2021-09-23 13:09:07      INR    499              IND   
21   R181106661 2021-09-23 13:09:50      INR    499              IND   
254  R798254987 2021-04-27 09:38:07      INR    321              IND   
292  R299286513 2021-05-04 17:20:28      INR    999              IND   

    Shipping City Shipping State Payment Method  \
15       VARANASI          IN-UP            NaN   
18          Delhi          IN-DL            NaN   
21          Delhi          IN-DL            NaN   
254     Berhampur          IN-OR            NaN   
292     Hyderabad          IN-AP            NaN   

                                          Product Name  Quantity  Year  
15   ONLINE TUITION FOR STD 4 and STD 5(FREE 5 DAYS...         1  2021  
18   Fun Activities for Kids LKG to STD 10(Singing,...         1  2021  
21   Fun Activities for Kids LKG to STD 10(Singing,...         1  2021  
254                 Mutton Curry Cut 500gm (Berhampur)         1  2021  
292  REALME BUDS PLUS Wireless Bluetooth Ears Buds ...         1  2021

In [36]:
yearly_sale = year_sale_data.groupby(['Year'], as_index=False)['Total'].mean()
yearly_sale

Year        Total
0  2016   707.142857
1  2017  1123.000000
2  2018  9163.448819
3  2019  6571.983333
4  2020  1419.833333
5  2021  2234.250000

In [37]:
yearly_sale['Total'] = yearly_sale['Total'].round(1)

In [38]:
yearly_sale

Year   Total
0  2016   707.1
1  2017  1123.0
2  2018  9163.4
3  2019  6572.0
4  2020  1419.8
5  2021  2234.2

Showing the results through visualization

In [39]:
year_fig = px.line(data_frame=yearly_sale, x='Year', y='Total', title='Average Sale by Year', labels={'Total':'Average Sale'},
        text='Total', markers=True)
year_fig.show()

##### Result - As per the result sale was high in year 2018

##### 3) Different Payment Methods used by Customers

In [40]:
data.head()

Order No DateTime Stamp Currency Total Shipping Country Shipping City  \
0  R121113121            NaN      INR  2299              IND    Chandrapur   
1  R472890631            NaN      INR  2299              IND    Chandrapur   
2  R004476488            NaN      INR  2299              IND    Chandrapur   
3  R526038353            NaN      INR  2299              IND    Chandrapur   
4  R658530771            NaN      INR   349              IND     NEW DELHI   

  Shipping State Payment Method  \
0          IN-MH            NaN   
1          IN-MH            NaN   
2          IN-MH            NaN   
3          IN-MH            NaN   
4          IN-DL            NaN   

                                        Product Name  Quantity  
0  Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
1  Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
2  Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
3  Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
4                                 Barbie Doll (pink)         1

In [41]:
pay_method_data= data[data['Payment Method'].notnull()]
pay_method_data.head()

Order No DateTime Stamp Currency Total Shipping Country Shipping City  \
26   R113535804            NaN      INR   999              IND     Dombivali   
137  R913187467            NaN      INR   219              IND      Kakinada   
138  R204461089            NaN      INR  1499              IND      Dindigul   
146  R602350202            NaN      INR   349              IND     Bengaluru   
147  R760464983            NaN      INR   799              IND         NOIDA   

    Shipping State      Payment Method  \
26           IN-MH    CCAvenue ₹999.00   
137          IN-AP    CCAvenue ₹219.00   
138          IN-TN  CCAvenue ₹1,499.00   
146          IN-KA    CCAvenue ₹349.00   
147          IN-UP    CCAvenue ₹799.00   

                                          Product Name  Quantity  
26                                 Lava Captain N1 Red         1  
137           Fresh Mushrooms Button 200gm (Berhampur)         1  
138           Vmax HX 750 Quadcopter Drone (No Camera)         1  
146  Boat Super EXTRA BASS HEADS 225 WIRED WITH MIC...         1  
147  Solanki King Car Bugatti Style Steering Radio ...         1

In [42]:
pay_method_data['Payment Method'].unique()

array(['CCAvenue ₹999.00', 'CCAvenue ₹219.00', 'CCAvenue ₹1,499.00',
       'CCAvenue ₹349.00', 'CCAvenue ₹799.00', 'CCAvenue ₹150.00',
       'CCAvenue ₹269.00', 'CCAvenue ₹399.00',
       'Offline Payment ₹8,999.00', 'Offline Payment ₹799.00',
       'Offline Payment ₹8,999.00;Offline Payment -₹8,999.00',
       'CCAvenue ₹1,798.00', 'CCAvenue ₹1,299.00',
       'Offline Payment ₹2,299.00', 'CCAvenue ₹400.00',
       'CCAvenue ₹499.00', 'CCAvenue ₹140.00', 'CCAvenue ₹1,700.00',
       'CCAvenue ₹1,599.00', 'CCAvenue ₹6,999.00',
       'Offline Payment ₹1,099.00', 'Offline Payment ₹1,999.00',
       'Offline Payment ₹299.00', 'Offline Payment ₹549.00',
       'Offline Payment ₹5,999.00',
       'Offline Payment ₹1,099.00;Offline Payment -₹1,099.00',
       'CCAvenue ₹1,099.00', 'Offline Payment ₹349.00',
       'Offline Payment ₹199.00', 'Offline Payment ₹400.00',
       'Offline Payment ₹1,199.00', 'Offline Payment ₹259.00',
       'Offline Payment ₹5,097.00', 'Offline Payment ₹230.0

We can see that her is two type of payment method and values are mixed up with price

Using the where function of numpy to change the values.

In [43]:
pay_method_data['Payment Method'] = np.where(pay_method_data['Payment Method'].str.contains('CCAven'), "CCAvenue", "Offline Payment")

C:\Users\Admin\AppData\Local\Temp\ipykernel_11140\3840913070.py:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [44]:
pay_method_data['Payment Method'].unique()

array(['CCAvenue', 'Offline Payment'], dtype=object)

Handling Duplicates in dataset

In [45]:
pay_method_data.duplicated().sum()

2

In [46]:
pay_method_data.drop_duplicates()
pay_method_data.head()

Order No DateTime Stamp Currency Total Shipping Country Shipping City  \
26   R113535804            NaN      INR   999              IND     Dombivali   
137  R913187467            NaN      INR   219              IND      Kakinada   
138  R204461089            NaN      INR  1499              IND      Dindigul   
146  R602350202            NaN      INR   349              IND     Bengaluru   
147  R760464983            NaN      INR   799              IND         NOIDA   

    Shipping State Payment Method  \
26           IN-MH       CCAvenue   
137          IN-AP       CCAvenue   
138          IN-TN       CCAvenue   
146          IN-KA       CCAvenue   
147          IN-UP       CCAvenue   

                                          Product Name  Quantity  
26                                 Lava Captain N1 Red         1  
137           Fresh Mushrooms Button 200gm (Berhampur)         1  
138           Vmax HX 750 Quadcopter Drone (No Camera)         1  
146  Boat Super EXTRA BASS HEADS 225 WIRED WITH MIC...         1  
147  Solanki King Car Bugatti Style Steering Radio ...         1

In [47]:
count_method = pay_method_data.groupby(['Payment Method'], as_index=False)['Order No'].count()
count_method

Payment Method  Order No
0         CCAvenue        60
1  Offline Payment       199

In [243]:
payment_fig = px.bar(count_method, x='Payment Method', y='Order No', color="Payment Method", title='Different Payment Methods',
        text_auto=True, labels={'Order No':'Number of Orders'})
payment_fig.show()

##### Result - Here we can see that mostly payment method used by customers is Offline Payment

##### 4) Top Products with high sales according to Payment Method.

In [49]:
data.isnull().sum()

Order No               0
DateTime Stamp      2801
Currency               0
Total                  0
Shipping Country       0
Shipping City         19
Shipping State        22
Payment Method      2780
Product Name           0
Quantity               0
dtype: int64

In [50]:
def new_func():
    return data['Product Name'].unique()
print(list[new_func()])

list[array(['Falcon Drone Four Axis Aircraft with 2.4 GHz RC, Blade Guard, Headless Mode LED Without Camera',
       'Barbie Doll (pink)',
       'Sony MDR-G45LP On-Ear Street Wired Headphone',
       'Reliance Jio Phone 2',
       'QUECHUA Ultra Compact Travel 10 Litre Backpack (Blue)',
       'Lava Captain N1 Red',
       'Zebronics ZEB-16A 16 inch (39.6 cm) LED Monitor Full HD',
       'Spider Man Mini Drone',
       'Candytech SPEEDO Dual USB Charger with 2.6 Amp Power',
       'ONLINE TUITION FOR STD 4 and STD 5(FREE 5 DAYS DEMO CBSE)',
       'Samsung Earphones With Mic Tuned by AKG (Black)',
       'Vmax HX 750 Quadcopter Drone (No Camera)',
       'Fun Activities for Kids LKG to STD 10(Singing, Dancing, Drawing, Craft, Music, Yoga, Karate)',
       'JBL A005 Portable Led Lightning Wireless Bluetooth Speaker',
       'Quarter Chicken Biryani (Chennai)', 'Boat BASS HEADS 152 (Black)',
       'Fresh Paneer 500gm (Berhampur)',
       'boAT Airdopes X3 Bass Wireless Earbuds with Cha

In [51]:
product_sale = data.copy()

In [52]:
product_sale.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3039 entries, 0 to 3038
Data columns (total 10 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Order No          3039 non-null   object
 1   DateTime Stamp    238 non-null    object
 2   Currency          3039 non-null   object
 3   Total             3039 non-null   object
 4   Shipping Country  3039 non-null   object
 5   Shipping City     3020 non-null   object
 6   Shipping State    3017 non-null   object
 7   Payment Method    259 non-null    object
 8   Product Name      3039 non-null   object
 9   Quantity          3039 non-null   int64 
dtypes: int64(1), object(9)
memory usage: 237.5+ KB


Handling the duplicate values in the dataset

In [53]:
product_sale.duplicated().sum()

3

In [54]:
product_sale[product_sale.duplicated()]

Order No             DateTime Stamp Currency  Total Shipping Country  \
2558  R754403637  29-03-2019 00:59:45 +0530      INR  44970              IND   
2614  R446418996  06-01-2019 01:33:48 +0530      INR  46166              IND   
2714  R754510307                        NaN      INR  36768              IND   

     Shipping City Shipping State              Payment Method  \
2558      Varanasi          IN-UP  Offline Payment ₹44,970.00   
2614      Varanasi          IN-UP  Offline Payment ₹46,166.00   
2714      Varanasi          IN-UP                         NaN   

                                           Product Name  Quantity  
2558  Vmax HX 750 Remote Control Flying Drone No Camera        15  
2614  Vmax HX 750 Remote Control Flying Drone No Camera         5  
2714       HX 750 Remote Control Flying Drone No Camera         8

In [55]:
product_sale.drop_duplicates(inplace=True)

In [56]:
product_sale.dtypes

Order No            object
DateTime Stamp      object
Currency            object
Total               object
Shipping Country    object
Shipping City       object
Shipping State      object
Payment Method      object
Product Name        object
Quantity             int64
dtype: object

In [57]:
product_sale['Total'].unique()

array(['2299', '349', '899', '3199', '598', '999', '3499', '1499', '299',
       '1000', '499', '63', '169', '150', '399', '1999', '1299', '599',
       '699', '300', '1899', '799', '4999', '1', '379', '4399', '549',
       '31999', '7800', '400', '369', '1199', '559', '199', '792', '219',
       '279', '1798', '8999', '450', '469', '1300', '2798', '1399', '130',
       '2999', '269', '429', '321', '949', '389', '584', '49', '798',
       '3495', '19999', '5499', '2199', '20', '0', '3198', '5949', '2',
       '3000', '7999', '21', '7998', '1449', '280', '7949', '140', '1699',
       '669', '1098', '698', '1398', '1598', '101', '649', '749', '6499',
       '2495', '160', '14098', '19949', '14949', '3999', '11997', '2249',
       '449', '1149', '149', '350', '800', '5999', '1249', '1998', '2049',
       '750', '7,18,281.00', '1649', '498', '3498', '330', '1748', '897',
       '1698', '30870', '3449', '1548', '1700', '1650', '3149', '3350',
       '13600', '34000', '3400', '17000', '6398'

In [58]:
product_sale[product_sale['Total'].str.contains(',')]

Order No DateTime Stamp Currency        Total Shipping Country  \
879   R765288033            NaN      INR  7,18,281.00              IND   
1434  R078719573            NaN      INR  1,69,950.00              IND   
3002  R121638056            NaN      INR  9,49,050.00              IND   
3003  R121638056            NaN      INR  9,49,050.00              IND   
3021  R168504447            NaN      INR  1,00,000.00              IND   

     Shipping City Shipping State Payment Method  \
879       Bhilwara          IN-RJ            NaN   
1434        Mumbai          IN-MH            NaN   
3002          test          IN-AN            NaN   
3003          test          IN-AN            NaN   
3021       CHENNAI          IN-TN            NaN   

                                           Product Name  Quantity  
879                           Cash On Delivery Services       999  
1434                                 Reliance Jio Phone       100  
3002  Bose Quiet Comfort 20 Acoustic Noise Cancellin...       999  
3003                            AmbranePowerBank P-1310       999  
3021                  Electric Steam Cooker Maestro MC1        50

Replacing the ',' with empty string and changing the datatype

In [59]:
product_sale['Total']=product_sale['Total'].replace('\D', '', regex=True).astype(int)

In [60]:
product_sale.dtypes

Order No            object
DateTime Stamp      object
Currency            object
Total                int32
Shipping Country    object
Shipping City       object
Shipping State      object
Payment Method      object
Product Name        object
Quantity             int64
dtype: object

In [100]:
product_sale = product_sale[product_sale['Shipping Country'] == 'IND']

Filtering the values which are not null in the Payment Method

In [141]:
product_sale = product_sale[product_sale['Payment Method'].notnull()]

In [142]:
product_sale.shape

(257, 10)

Changing the payment method into the right format

In [143]:
product_sale['Payment Method'] = np.where(product_sale['Payment Method'].str.contains('CCAven'), 'CCAvenue', 'Offline Payment')

C:\Users\Admin\AppData\Local\Temp\ipykernel_11140\1359200798.py:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [150]:
product_sale['Payment Method'].unique()

array(['CCAvenue', 'Offline Payment'], dtype=object)

Applying the Groupby method in the required columns

In [154]:
top_products = product_sale.groupby(['Product Name', 'Payment Method'], as_index=False)['Total'].mean()
top_products.head()

Product Name   Payment Method   Total
0                                1KG Chicken Biryani         CCAvenue   100.0
1                      1KG Chicken Biryani (Chennai)         CCAvenue   145.0
2                     Auto ePass 2003 FIPS USB Token  Offline Payment   650.0
3  Avengers Captain America Four Axis Aircraft Dr...  Offline Payment  2299.0
4  BOSE Sound True SQ2  Headphones (White,Black,Red)         CCAvenue   300.0

Keeping Top 10 Products with their payment method

In [160]:
top_products = top_products.sort_values(by='Total', ascending=False)
top_products = top_products.head(10)
top_products

Product Name   Payment Method  \
28     Drone Quadcopter 2.4G 6-Channel Without Camera         CCAvenue   
41                         HX770 V-Max Aircraft Drone  Offline Payment   
38                   HX-713 Remote Control Helicopter  Offline Payment   
60  MitSonic 16 inches (40cm) FULL HD Gorilla Glas...  Offline Payment   
36       HX 750 Remote Control Flying Drone No Camera  Offline Payment   
59                                 Minions Mini Drone  Offline Payment   
42                         Hot Wheels Car 5 Gift Pack  Offline Payment   
94  The Amazing Spider Man Q series 2.4GHz RC Quad...  Offline Payment   
90                              Spider Man Mini Drone  Offline Payment   
62                   MitSonic 16 inches (40cm) LED TV  Offline Payment   

           Total  
28  1.012837e+06  
41  2.832464e+05  
38  2.327492e+05  
60  2.210010e+05  
36  1.397341e+05  
59  3.956217e+04  
42  3.851900e+04  
94  3.851900e+04  
90  2.762944e+04  
62  2.299500e+04

In [161]:
top_products['Total'] = top_products['Total'].round(1)
top_products.reset_index(drop=True, inplace=True)
top_products.head()

Product Name   Payment Method  \
0     Drone Quadcopter 2.4G 6-Channel Without Camera         CCAvenue   
1                         HX770 V-Max Aircraft Drone  Offline Payment   
2                   HX-713 Remote Control Helicopter  Offline Payment   
3  MitSonic 16 inches (40cm) FULL HD Gorilla Glas...  Offline Payment   
4       HX 750 Remote Control Flying Drone No Camera  Offline Payment   

       Total  
0  1012837.0  
1   283246.4  
2   232749.2  
3   221001.0  
4   139734.1

Now, Visualizing the insights for better data understanding

In [169]:
product_fig = px.bar(top_products, x='Product Name', y='Total', color='Payment Method', labels={'Total':'Prices'},
        title='Top 10 Products with Different Payment Methods', text = 'Total')
product_fig.show()

##### Result - Here we can see Drone Quadcopter has given the top contribution in the sale price

##### 5) Which state has contributed most in the sale.

In [170]:
data.head()

Order No DateTime Stamp Currency Total Shipping Country Shipping City  \
0  R121113121            NaN      INR  2299              IND    Chandrapur   
1  R472890631            NaN      INR  2299              IND    Chandrapur   
2  R004476488            NaN      INR  2299              IND    Chandrapur   
3  R526038353            NaN      INR  2299              IND    Chandrapur   
4  R658530771            NaN      INR   349              IND     NEW DELHI   

  Shipping State Payment Method  \
0          IN-MH            NaN   
1          IN-MH            NaN   
2          IN-MH            NaN   
3          IN-MH            NaN   
4          IN-DL            NaN   

                                        Product Name  Quantity  
0  Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
1  Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
2  Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
3  Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
4                                 Barbie Doll (pink)         1

In [172]:
state_sale = data.copy()
state_sale.head()

Order No DateTime Stamp Currency Total Shipping Country Shipping City  \
0  R121113121            NaN      INR  2299              IND    Chandrapur   
1  R472890631            NaN      INR  2299              IND    Chandrapur   
2  R004476488            NaN      INR  2299              IND    Chandrapur   
3  R526038353            NaN      INR  2299              IND    Chandrapur   
4  R658530771            NaN      INR   349              IND     NEW DELHI   

  Shipping State Payment Method  \
0          IN-MH            NaN   
1          IN-MH            NaN   
2          IN-MH            NaN   
3          IN-MH            NaN   
4          IN-DL            NaN   

                                        Product Name  Quantity  
0  Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
1  Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
2  Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
3  Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
4                                 Barbie Doll (pink)         1

Handling the duplicates in the dataset

In [174]:
state_sale.duplicated().sum()

3

In [175]:
state_sale.drop_duplicates()

Order No             DateTime Stamp Currency Total Shipping Country  \
0     R121113121                        NaN      INR  2299              IND   
1     R472890631                        NaN      INR  2299              IND   
2     R004476488                        NaN      INR  2299              IND   
3     R526038353                        NaN      INR  2299              IND   
4     R658530771                        NaN      INR   349              IND   
...          ...                        ...      ...   ...              ...   
3034  R243395005                        NaN      INR    50              IND   
3035  R607209508                        NaN      INR   500              IND   
3036  R140316623                        NaN      INR  4000              IND   
3037  R192037084                        NaN      INR  2000              IND   
3038  R414254148  28-09-2016 19:05:30 +0530      INR  2000              IND   

     Shipping City Shipping State             Payment Method  \
0       Chandrapur          IN-MH                        NaN   
1       Chandrapur          IN-MH                        NaN   
2       Chandrapur          IN-MH                        NaN   
3       Chandrapur          IN-MH                        NaN   
4        NEW DELHI          IN-DL                        NaN   
...            ...            ...                        ...   
3034         Gccch          IN-AN                        NaN   
3035     BERHAMPUR          IN-OR                        NaN   
3036     berhampur          IN-OR                        NaN   
3037          sdas          IN-AN                        NaN   
3038          sdas          IN-AN  Offline Payment ₹2,000.00   

                                           Product Name  Quantity  
0     Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
1     Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
2     Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
3     Falcon Drone Four Axis Aircraft with 2.4 GHz R...         1  
4                                    Barbie Doll (pink)         1  
...                                                 ...       ...  
3034                 Ear Wired Earphones With Mic White         1  
3035                                  I Kall K-66 White         1  
3036                  Electric Steam Cooker Maestro MC1         2  
3037                  Electric Steam Cooker Maestro MC1         1  
3038                  Electric Steam Cooker Maestro MC1         1  

[3036 rows x 10 columns]

In [183]:
state_sale = state_sale[state_sale['Shipping Country'] == 'IND']

In [192]:
state_sale.duplicated(subset=['Order No']).sum()

67

In [195]:
state_sale = state_sale.drop_duplicates(subset=['Order No'])
state_sale.shape

(2932, 10)

In [199]:
state_sale.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2932 entries, 0 to 3038
Data columns (total 10 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Order No          2932 non-null   object
 1   DateTime Stamp    207 non-null    object
 2   Currency          2932 non-null   object
 3   Total             2932 non-null   object
 4   Shipping Country  2932 non-null   object
 5   Shipping City     2913 non-null   object
 6   Shipping State    2932 non-null   object
 7   Payment Method    228 non-null    object
 8   Product Name      2932 non-null   object
 9   Quantity          2932 non-null   int64 
dtypes: int64(1), object(9)
memory usage: 252.0+ KB


In [201]:
state_sale = state_sale[~state_sale['Total'].str.contains(',')]

In [202]:
state_sale.shape

(2928, 10)

Converting the 'Total' data type into int after float

In [211]:
state_sale['Total'] = state_sale['Total'].astype(float)

In [212]:
state_sale['Total'] = state_sale['Total'].astype(int)

In [213]:
state_sale.dtypes

Order No            object
DateTime Stamp      object
Currency            object
Total                int32
Shipping Country    object
Shipping City       object
Shipping State      object
Payment Method      object
Product Name        object
Quantity             int64
dtype: object

In [225]:
top_state_sale = state_sale.groupby(['Shipping State'], as_index=False)['Total'].mean()
top_state_sale.head()

Shipping State        Total
0          IN-AN  2550.496599
1          IN-AP  1802.073446
2          IN-AR  1699.000000
3          IN-AS  2772.027027
4          IN-BR  2080.910569

In [226]:
top_state_sale.sort_values(ascending=False, by='Total', inplace=True)

In [228]:
top_state_sale['Total']=top_state_sale['Total'].round(1)

In [229]:
top_state_sale.reset_index(drop=True, inplace=True)

In [237]:
top_state_sale = top_state_sale.head(10)
top_state_sale

Shipping State    Total
0          IN-SK  31999.0
1          IN-NL   5761.8
2          IN-CH   5260.4
3          IN-HP   4876.2
4          IN-PY   4290.9
5          IN-UP   3938.1
6          IN-JK   3078.3
7          IN-MP   3050.0
8          IN-UT   2906.8
9          IN-AS   2772.0

Replacing the states code into their standard original names

In [238]:
top_state_sale['Shipping State'].replace({'IN-SK':'Sikkim', 'IN-NL':'Nagaland', 'IN-CH':'Chandigarh', 'IN-HP':'Himanchal Pradesh',
                'IN-PY':'Puducherry', 'IN-UP':'Uttar Pradesh', 'IN-JK':'Jammu and Kashmir', 'IN-MP':'Madhya Pradesh', 'IN-UT':'Uttarakhand',
                'IN-AS':'Assam'}, inplace=True)

Visualizing the data for better understanding

In [241]:
px.bar(top_state_sale, x='Shipping State', y='Total', text='Total', labels={'Total':'Avg_Sale', 'Shipping State':'State'},
        title='Top 10 States By Average Sale', color='Shipping State')

##### Result - Sikkim State has the most contribution in the growing of the company's sale.